core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -353,7 +355,28 @@ public static boolean isTablesEnumeratingTable(InformationSchemaTable table)
                     .flatMap(prefix -> tables.get().stream()
                             .filter(this::isLowerCase)
                             .map(table -> new QualifiedObjectName(catalogName, prefix.getSchemaName().get(), table)))
-                    .filter(objectName -> !isColumnsEnumeratingTable(informationSchemaTable) || metadata.getTableHandle(session, objectName).isPresent() || metadata.getView(session, objectName).isPresent())
+                    .filter(objectName -> {
+                        if (isColumnsEnumeratingTable(informationSchemaTable)) {
+                            if (metadata.getView(session, objectName).isPresent()) {
+                                return true;
+                            }
+                            try {
```

Comment:
IMO extracting it to a separate method may not be very useful, because exception catching here is context dependent, and seems easier to read inline. wdyt?